packages/odyssey-react-mui/src/components/Banner/Banner.tsx

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { AlertColor } from "@mui/material";
+import React, { useState } from "react";
+import { Alert, Link } from "../..";
+
+export interface BannerProps {
+  /**
+   * Determine the color and icon of the alert
+   */
+  severity?: AlertColor;
+  /**
+   * Sets the ARIA role of the alert
+   * ("status" for something that dynamically updates, null for something
+   * unchanging)
+   */
+  role?: "status" | undefined;
+  /**
+   * The text content of the alert
+   */
+  children: string;
+  /**
+   * If true, the alert can be closed with the X icon
+   */
+  isDismissable?: boolean;
+  /**
+   * If defined, the alert will include a link to the URL
+   */
+  linkUrl?: string;
+  /**
+   * If linkUrl is not undefined, this is the text of the link.
+   * If left blank, it defaults to "Learn more".
+   * Note that linkText does nothing if linkUrl is not defined
+   */
+  linkText?: string;
+}
+
+export const Banner = ({
+  isDismissable = true,
+  severity = "info",
+  role,
+  linkUrl,
+  linkText = "Learn more",
+  children,
+}: BannerProps) => {
+  const [isVisible, setIsVisible] = useState(true);
+
+  const handleClose = () => setIsVisible(false);
+
+  return (
+    <React.Fragment>
+      {isVisible && (
+        <Alert
+          role={role}
+          variant="banner"
+          severity={severity}
+          onClose={isDismissable ? handleClose : undefined}
+        >
```

Comment:
I really don't think we should be handling the Alert visibility in the alert itself, it should be handled by the renderer.

This is a cross-pollination of roles. Typically, consumers should handle the rendering of components, not the components themselves. You can create a hook with the logic necessary for consuming it.

There are better ways of doing this with a children-first pattern I've talked about, but that's not a pattern we use at Okta, and it would require a more diverse set of tools rather than per-component logic.

At this point, I'd recommend making visibility something handled outside of `Banner`. You can provide a hook outside of it. You'll also want to add a callback for when a dismissal occurred so the logic outside can dismiss the banner.

Either you or someone else can add the necessary hook with visibility logic. It could be `useBanner`, but it'd best be something generic like `useVisibility`. I've written a hook like this already in my open-source library of Children-First Components: https://github.com/Sawtaytoes/children-first-components/blob/master/src/useVisibility.ts.

We don't have to take it verbatim, but we should discuss possible solutions rather than building hide-show logic into every Odyssey component.

Another option is following MUI's take and doing it however they do it, but there's nothing stopping MUI from bad-design patterns that, because of copy-paste-modify, will be replicated throughout our apps at Okta.

---

Lastly, we need to change this hide-show functionality because you have no way of telling the consumer when to remove the banner or how to show it again.

Right now, it gets rendered when an error occurs and then hides itself, but the consumer still has it rendered. If another error occurs, it won't show because it's already rendered, and the consumer has no way of passing a prop saying "show again".

Reply:
This is excellent feedback, thanks @KevinGhadyani-Okta! I don't think I have the chops to add the logic hook myself, so I'll leave that to you or Shred. And again, I just want to be helpful, so if it's counterproductive to have me touching code myself please let me know and I'm happy to leave it to those who actually know what they're doing — I don't want to unintentionally make _more_ work for you by pretending to be an engineer 😅

Reply:
MUI exposes `onClose` and accepts any function; if a function is passed, it displays the UI appropriate for dismissal. I think that aligns with your suggestion above without us having to solve for _how_ things are dismissed in various apps.

If that's the case, our stories will need some example code to demo the functionality, but the component itself could ship without an opinion.

<img width="826" alt="Screen Shot 2023-01-12 at 10 48 39 AM" src="https://user-images.githubusercontent.com/36284167/212153295-e19b2b72-5f2c-428c-bf71-c04f9a6a061f.png">